store/src/main/java/org/apache/rocketmq/store/DefaultMessageStore.java

```diff
@@ -948,24 +952,12 @@ public QueryMessageResult queryMessage(String topic, String key, int maxNum, lon
                 long offset = queryOffsetResult.getPhyOffsets().get(m);
 
                 try {
-
-                    boolean match = true;
```

Comment:
Actually match isn't used in broker, so how about move the check logic to client？

Reply:
OK, I'll change it soon